### PR TITLE
A scrollViewer

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
This scroll viewer allows containers to be added to the viewer window using scrollViewer.addWindow.

Added containers keep their size as given with respect to the host window except for a TextBlock. A TextBlock added to the viewer window will be adjusted so that its width matches that of the viewing window and its height adjusted to that needed to show all lines. To fix the size of a TextBlock it needs to be put inside a container and the container added to the window. 

This PR needs to be checked by Deltakosh as he has some ideas about how a scrollViewer should be implemented and this version might not match his expectations.

### CODE REVIEW

**Comment by BabylonJSGuide** [comment]
The PGs I used to check locally can be found at https://drive.google.com/drive/folders/1YySEFMSL_Oj1JoNtfuUk_uw28d7IdsJv?usp=sharing

File: src/Actions/babylon.action.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ module BABYLON {
      * The action to be carried out following a trigger
      * @see http://doc.babylonjs.com/how_to/how_to_use_actions#available-actions
      */
+
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply src/Actions/babylon.action.ts]: Not sure what happened here will delete

File: gui/src/2D/controls/scrollViewer.ts [hunk 2]
```diff
@@ -0,0 +1,364 @@
+import { Measure } from "../measure";
+import { Rectangle } from "./rectangle";
+import { Grid } from "./grid";
+import { Control } from "./control";
+import { Slider } from "./slider";
+import { ValueAndUnit } from "../valueAndUnit";
+import { Container } from "./container";
+import { TextBlock } from "./textBlock";
+
+/** Class used to hold a viewer window and sliders in a grid
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply gui/src/2D/controls/scrollViewer.ts]: Done

File: gui/src/2D/controls/scrollViewer.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    private _scrollGridWidth: number = 30;
+    private _scrollGridHeight: number = 30;
+    private _widthScale: number;
+    private _heightScale: number;
+    private _endLeft: number;
+    private _endTop: number;
+    private _window: Container;
+    private _windowContents: Control;
+
+    /** Adds windowContents to the grid view window 0, 0
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply gui/src/2D/controls/scrollViewer.ts]: Done

File: gui/src/2D/controls/scrollViewer.ts [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+     * @param windowContents the contents to add the grid view window 0, 0
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply gui/src/2D/controls/scrollViewer.ts]: 0, 0 removed

File: gui/src/2D/controls/scrollViewer.ts [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    /**
+    * Creates a new ScrollViewer
+    * @param name of ScrollViewer
+    */
+    constructor(
+        /** name of ScrollViewer */
+        public name?: string) {
+        super(name);
+
+        //this.clipChildren = false;
```
[the remark this answers is not in the document]
Comment (BabylonJSGuide) [reply gui/src/2D/controls/scrollViewer.ts]: removed

**Comment by dad72** [comment]
I did not find the documentation for this newcomer. Is it planned? It would be nice to have the documentation for this scrollView with a PG example.

Thank you for this new container.

**Comment by BabylonJSGuide** [comment]
In the middle of writing it as you message me :)

**Pull request merged**